### PR TITLE
Add language option to gatsby-source-wikipedia plugin

### PR DESCRIPTION
**Pull request by Steven24K**
## Description

I have added a language option to the gatsby-source-wikipedia plugin. When you look at the Wikipedia urls then you notice the language subdomain. For many international subjects there exists a translated article. For instance: https://de.wikipedia.org/wiki/Justin_Bieber, https://nl.wikipedia.org/wiki/Justin_Bieber, https://fr.wikipedia.org/wiki/Justin_Bieber or https://en.wikipedia.org/wiki/Justin_Bieber

All point to the same subject, but in a different language. 

### Documentation

I have made the lang parameter optional, to prevent this plugin from not working in existing projects. When left empty it defaults to the English wikipedia.

### CODE REVIEW

File: packages/gatsby-source-wikipedia/src/fetch.js [hunk 1]
```diff
@@ -2,13 +2,13 @@ const Promise = require(`bluebird`)
...
 
-const fetchNodesFromSearch = ({ query, limit = 15 }) =>
-  search({ query, limit }).then(results =>
+const fetchNodesFromSearch = ({ query, limit = 15, lang = `en` }) =>
+  search({ query, limit, lang }).then(results =>
     Promise.map(results, async (result, queryIndex) => {
-      const rendered = await getArticle(result.id)
-      const metadata = await getMetaData(result.id)
+      const rendered = await getArticle(result.id)(lang)
+      const metadata = await getMetaData(result.id)(lang)
```
Comment (vladar) [inline packages/gatsby-source-wikipedia/src/fetch.js:11]:
```suggestion
      const rendered = await getArticle(result.id, lang)
      const metadata = await getMetaData(result.id, lang)
```

File: packages/gatsby-source-wikipedia/src/fetch.js [hunk 2]
```diff
@@ -70,16 +70,16 @@ const search = ({ query, limit }) =>
...
           description: descriptions[i],
-          id: /en.wikipedia.org\/wiki\/(.+)$/.exec(urls[i])[1],
+          id: term,
         }
       })
     )
 
-const getArticle = name =>
-  fetch(`https://en.m.wikipedia.org/wiki/${name}?action=render`)
+const getArticle = name => lang =>
```
Comment (vladar) [inline packages/gatsby-source-wikipedia/src/fetch.js:78]: There is no reason to use currying here. Just use two args:

```suggestion
const getArticle = (name, lang) =>
```

File: packages/gatsby-source-wikipedia/src/fetch.js [hunk 3]
```diff
@@ -70,16 +70,16 @@ const search = ({ query, limit }) =>
         return {
           title,
           description: descriptions[i],
-          id: /en.wikipedia.org\/wiki\/(.+)$/.exec(urls[i])[1],
+          id: term,
```
Comment (vladar) [inline packages/gatsby-source-wikipedia/src/fetch.js:73]: Why this change? What `term` looks like?

File: packages/gatsby-source-wikipedia/src/fetch.js [hunk 4]
```diff
@@ -20,9 +20,9 @@ const fetchNodesFromSearch = ({ query, limit = 15 }) =>
     })
   )
 
-const getMetaData = name =>
+const getMetaData = name => lang =>
```
Comment (vladar) [inline packages/gatsby-source-wikipedia/src/fetch.js:23]: No reason to use currying here:

```suggestion
const getMetaData = (name, lang) =>
```

File: packages/gatsby-source-wikipedia/src/fetch.js [hunk 5]
```diff
@@ -70,16 +70,16 @@ const search = ({ query, limit }) =>
...
     )
 
-const getArticle = name =>
-  fetch(`https://en.m.wikipedia.org/wiki/${name}?action=render`)
+const getArticle = name => lang =>
+  fetch(`https://${lang}.m.wikipedia.org/wiki/${name}?action=render`)
     .then(res => res.text())
     .then(pageContent =>
-      pageContent.replace(/\/\/en\.wikipedia\.org\/wiki\//g, `/wiki/`)
+      pageContent.replace(`/\/\/${lang}\.wikipedia\.org\/wiki\//g`, `/wiki/`)
```
Comment (vladar) [inline packages/gatsby-source-wikipedia/src/fetch.js:82]: I don't think this will do what you expect. If we don't care about the language value in replacement, then this should be good:
```suggestion
      pageContent.replace(/\/\/[a-z]+\.wikipedia\.org\/wiki\//g, `/wiki/`)
```

But that also means that links in different languages will point to the same URL. This works when you deal with a single language. But won't work that well for multilanguage setups.

I assume a multilanguage setup is a rare case and we can address it separately if someone actually needs it.

**Review by vladar — CHANGES_REQUESTED** [review]

**Comment by LekoArts** [comment]
Hi,

this PR became stale so I'm closing it for now. There's also a similar PR (https://github.com/gatsbyjs/gatsby/pull/28637) so if you (or someone else) wants to pick this up again please see both review comments. Thanks!
